trinity/protocol/common/proto.py

```diff
@@ -0,0 +1,53 @@
+
+
+class ChainProtocol(Protocol, ABC):
+
+    @classmethod
+    def configure_protocol(
+            cls,
+            headerdb: BaseAsyncHeaderDB,
+            network_id: int,
+            vm_configuration: Tuple[Tuple[int, Type[BaseVM]], ...]) -> Type[ChainProtocol]:
+        return type(
```

Comment:
I wouldn't say it is definitive, but I'm closing in on this being a pattern we try to avoid.  However.... I did struggle a decent bit to avoid this in my previous refactor of the peer class and repeatedly almost went this route as it made things easier....

That said, any way we can avoid dynamic class creation is probably going to save us some type hinting headaches.